### PR TITLE
Add concourse pipeline to deploy apps to ECS

### PR DESCRIPTION
**Pull request by bilbof**
This is a demo example to start deploying apps to ECS through a Concourse pipeline.

(VPN access + GitHub production login required): https://cd.gds-reliability.engineering/teams/govuk-tools/pipelines/govuk-cd-frontend:

Screenshot 2020-10-06 at 18 18 21

Screenshot 2020-10-06 at 18 18 37

More work is required to:

- add working smoke tests, and other behaviour described in proposal 005
- replace ruby stuff with a concourse ECS plugin
- build and push Docker image to DockerHub in CI

These changes can come in future commits.

Story: https://trello.com/c/C2ubxCmM/175-minimal-deployment-solution
Epic: https://trello.com/c/KBHxP0tt/180-deployments

### CODE REVIEW

File: concourse/pipelines/cd-frontend.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+jobs:
+- name: update-pipeline
+  plan:
+  - get: govuk-infrastructure
+    trigger: true
+  - file: govuk-infrastructure/concourse/pipelines/cd-frontend.yml
+    set_pipeline: govuk-cd-frontend
+- name: deploy
+  plan:
+  - get: govuk-infrastructure
```
Comment (tlwr) [inline concourse/pipelines/cd-frontend.yml:10]:
```suggestion
  - get: govuk-infrastructure
    passed: [update-pipeline]
```

If your change contains an update to the pipeline, and an update to the infrastructure, they should happen in a deterministic fashion.

Comment (bilbof) [reply concourse/pipelines/cd-frontend.yml:10]: Thanks!

File: concourse/pipelines/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  - name: publishing-api
+    jobs:
+      - deploy-publishing-api
+
+jobs:
+  - name: update-pipeline
+    plan:
+    - get: govuk-infrastructure
+      trigger: true
+    - file: govuk-infrastructure/concourse/pipelines/cd-frontend.yml
```
Comment (sengi) [inline concourse/pipelines/deploy.yml:89]: I couldn't find this file - am I missing something? :)

File: concourse/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#========================================================================================
+# This Dockerfile creates a container image with the dependencies required
+# to deploy
+# This is needed for running the deploy scripts in a container environment like Concourse.
+#========================================================================================
+FROM governmentpaas/awscli:latest
+
+RUN apk add --no-cache ruby-full=2.7.1-r3
```
Comment (sengi) [inline concourse/Dockerfile:8]: Do we have a solution for managing these dependency versions? (Dependabot?)

**Review by sengi — COMMENTED** [review]
Just a few fairly generic comments from me. I can't really comment as to the important stuff (the design) here as I'm not familiar with Concourse yet.

File: concourse/pipelines/deploy.yml (hunk 2 above)
Comment (bilbof) [reply concourse/pipelines/deploy.yml:89]: Thanks, fixed in superseding PR :)

**Comment by bilbof** [comment]
Superseded by https://github.com/alphagov/govuk-infrastructure/pull/40

@sengi thank you for your comments. I have addressed some of them in #40, I will address all of them in a followup PR for the Concourse pipeline.

**Pull request closed without merging**